pkg/controllers/managementuserlegacy/monitoring/clusterHandler.go

```diff
@@ -348,9 +355,33 @@ func (ch *clusterHandler) deployApp(appName, appTargetNamespace string, appProje
 		"prometheus.ruleSelector.matchExpressions[0].values[1]":                             monitoring.CattleMonitoringPrometheusRuleLabelValue,
 	}
 
-	appAnswers, appCatalogID, err := monitoring.OverwriteAppAnswersAndCatalogID(optionalAppAnswers, cluster.Annotations, ch.app.catalogTemplateLister, ch.cattleCatalogManager, ch.clusterName)
+	// build list of known keys that should NOT be forced to a string
+	// e.g. "true", "false", "123", "2379"
+	var knownUnforcedStringKeys []string
+
+	detectNotForcedStringKeys := func(answers map[string]string) []string {
+		var notForcedStringKeys []string
+		for k := range answers {
+			// all boolean keys must NOT be forced to be a string
+			if k == "true" || k == "false" {
+				notForcedStringKeys = append(notForcedStringKeys, k)
+			}
+			// if any default values are numbers, add a check here to ensure those number values are NOT forced as a string
+		}
+		return notForcedStringKeys
+	}
+
+	knownUnforcedStringKeys = append(knownUnforcedStringKeys, detectNotForcedStringKeys(optionalAppAnswers)...)
+	knownUnforcedStringKeys = append(knownUnforcedStringKeys, detectNotForcedStringKeys(mustAppAnswers)...)
```

Comment:
This section looks incomplete based on the comments. I don't follow why we expose the ability to force things but now we are deleting things a user tells us to force? And this is looking at keys? 

This code is also triplicated so if it's necessary we should create a helper somewhere

Reply:
Helper makes sense, didn't know where to put it so just need to determine that. I didn't want to impose the number-parsing overhead if not necessary, but it does make sense to add it in.

This functionality you're questioning is analogous to required App settings in the monitoring application. All the surrounding logic here relates to v1 monitoring and extra Helm answers, and specific Answers _must_ have certain values, so therefore certain Answers that are intended to be boolean or numbers _must_ be unable to be forced as a string.

Reply:
Just added d07994658e22b4e8a0943f81e7fd30aa62784cf5 that addresses the duplicated code and detects numbers as well.